### PR TITLE
Announcement that the snapshot branches are moving

### PR DESCRIPTION
**Pull request by carols10cents**
I'm hoping to get this reviewed and posted on Monday, 2022-02-14.

### CODE REVIEW

**Comment by Eh2406** [comment]
Will removing those branches mean that people have to download the entire index again if the squash has happened since last time they updated? When we did the first squash we tested and incremental downloads were still possible. I hope that's still true.

File: posts/2022-02-14-crates-io-snapshot-branches.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+release: false
+---
+
+Every so often, the [crates.io index](https://github.com/rust-lang/crates.io-index)'s Git history
+is [squashed into one
+commit](https://internals.rust-lang.org/t/cargos-crate-index-upcoming-squash-into-one-commit/8440)
+to minimize the history Cargo needs to download. When the index is squashed, we have saved snapshot
+branches to preserve the history of crate publishes.
+
+Those branches are using server resources because of the recompression of objects that are
```
Comment (shepmaster) [inline posts/2022-02-14-crates-io-snapshot-branches.md:14]:
```suggestion
to minimize the history Cargo needs to download. When the index is squashed, we save snapshots
to preserve the history of crate publishes.

Currently, those snapshots are stored as branches in the main index Git repository. This uses server resources because of the recompression of objects that are
```

File: posts/2022-02-14-crates-io-snapshot-branches.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+delta-compressed against objects that are only referenced in branches not fetched by the client. We
+will be deleting the snapshot branches to ensure that all objects referenced in the master branch
```
Comment (shepmaster) [inline posts/2022-02-14-crates-io-snapshot-branches.md:16]:
```suggestion
will be deleting the snapshot branches from this repository to ensure that all objects referenced in the master branch
```

**Comment by bjorn3** [comment]
> When the squashed branches where in repo, step 3 was O(1) not O(size of index). Is this still true?

It feels to me like this hasn't been true for quite a while. Not sure though.

File: posts/2022-02-14-crates-io-snapshot-branches.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+Every so often, the [crates.io index](https://github.com/rust-lang/crates.io-index)'s Git history
+is [squashed into one
+commit](https://internals.rust-lang.org/t/cargos-crate-index-upcoming-squash-into-one-commit/8440)
+to minimize the history Cargo needs to download. When the index is squashed, we have saved snapshot
+branches to preserve the history of crate publishes.
+
+Those branches are using server resources because of the recompression of objects that are
+delta-compressed against objects that are only referenced in branches not fetched by the client. We
```
Comment (pietroalbini) [inline posts/2022-02-14-crates-io-snapshot-branches.md:15]: This sentence is a bit hard to parse. Maybe we can simplify it a bit more?

> Those branches are using server resources though, as the server still has to consider their contents whenever Cargo asks for the `master` branch.

**Review by pietroalbini — COMMENTED** [review]
Looks good, thanks Carol for writing this! There's only one sentence that IMO needs a bit of tweaking.

**Comment by Eh2406** [comment]
> > When the squashed branches where in repo, step 3 was O(1) not O(size of index). Is this still true?

> It feels to me like this hasn't been true for quite a while. Not sure though.

The experiments were done before the first squash. That's what we observed at the time. If it had not been working that would also be important information for us to know. But we should collect the evidence, one way or the other.

**Comment by carols10cents** [comment]
Ok, I've made all suggested changes and I think this post is ready to go live tomorrow, so this is ready for someone to approve now.

**Comment by carols10cents** [comment]
Ping @rust-lang/core, could someone review this today please? 🙏🏻

File: posts/2022-02-14-crates-io-snapshot-branches.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+title: "Crates.io Index Snapshot Branches Moving"
+author: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>
```
Comment (badboy) [inline posts/2022-02-14-crates-io-snapshot-branches.md:4]: The blog doesn't render markdown for the author field.
This should be just the following:

```suggestion
author: The Crates.io Team
```

(that's used by at least one other blog post)

**Review by badboy — APPROVED** [review]
Minimal change in the metadata. Otherwise good to go.

On behalf of core: r+

**Comment by carols10cents** [comment]
I think someone from core actually has to click the merge button?
